### PR TITLE
Front concourse

### PR DESCRIPTION
**Pull request by thullSL**

### CODE REVIEW

File: concourse/resources/staticS3/resource/check [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#! /bin/bash
+
+echo []
```
Comment (csscottlogic) [inline concourse/resources/staticS3/resource/check]: Why is this here and not in the ```in``` script?

Comment (thullSL) [inline concourse/resources/staticS3/resource/check]: My interpretation of the [spec](https://concourse.ci/implementing-resources.html) was that check could return an empty list which we can do whereas in must return a version object which we can't as we don't have versions (we should probably have versions). 

I'm going to try check returning nothing as it may taken different ways given this from the spec
> The list may be empty, if the given version is already the latest.

File: concourse/resources/staticS3/resource/out [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#! /bin/bash
+
+node /code/bucketChucker $@
+echo '{ "version": { "ref": "'$BUILD_ID'" } }'
```
Comment (csscottlogic) [inline concourse/resources/staticS3/resource/out]: What does this do?

**Comment by csscottlogic** [comment]
We should probably think about versioning sooner rather than later. OK for now though.

File: concourse/resources/staticS3/code/bucketChucker.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        fs.readdir,
+        joinPaths(baseDir,subDir),
+        function(data) {return {base: baseDir, sub: subDir, files:data};},
+        function(data) {return "Read " + data.length + " files from build directory:  "  + data;}
+    );
+}
+
+function uploadDirectory(dir) {
+    console.error("Starting upload from:", dir.base, dir.sub);
+    var directoryProccessor = createDirectoryProccessor(dir);
```
Comment (csscottlogic) [inline concourse/resources/staticS3/code/bucketChucker.js]: "processor"

File: concourse/resources/staticS3/code/bucketChucker.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    return new Promise.all(dir.files.map(directoryProccessor));
+}
+
+function createDirectoryProccessor(dir) {
+    return function(fileName) {
+        return proccessDirectory(dir, fileName);
+    };
+}
+
+function proccessDirectory(dir, fileName) {
```
Comment (csscottlogic) [inline concourse/resources/staticS3/code/bucketChucker.js]: "process"

File: concourse/resources/staticS3/code/package.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "StaticS3Pusher",
+  "version": "0.0.1",
+  "dependencies": {
+    "aws-sdk": "~2.2.39",
+    "promise": "~7.1.1"
+  },
+  "devDependencies": {
```
Comment (csscottlogic) [inline concourse/resources/staticS3/code/package.json]: Shouldn't mime be a production dependency?

File: concourse/resources/staticS3/resource/out (hunk 2 above)
Comment (thullSL) [inline concourse/resources/staticS3/resource/out]: $BUILD_ID is passed in from other places, it was just to have something unique to give back in the correct format as per the spec

File: concourse/resources/staticS3/code/bucketChucker.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+function uploadDirectory(dir) {
+    console.error("Starting upload from:", dir.base, dir.sub);
+    var directoryProcessor = createDirectoryProcessor(dir);
+    return new Promise.all(dir.files.map(directoryProcessor));
+}
+
+function createDirectoryProcessor(dir) {
+    return function(fileName) {
+        return proccessDirectory(dir, fileName);
```
Comment (lclarksl) [inline concourse/resources/staticS3/code/bucketChucker.js]: "processDirectory"

**Comment by lclarksl** [comment]
:shipit:

**Comment by dwybourn** [comment]
:shipit:
